types/priv_validator.go

```diff
@@ -225,7 +227,13 @@ func (privVal *PrivValidatorFS) save() {
 		// `@; BOOM!!!
 		cmn.PanicCrisis(err)
 	}
-	err = cmn.WriteFileAtomic(privVal.filePath, jsonBytes, 0600)
+	outFile := privVal.filePath
+	dir := filepath.Dir(outFile)
+	if err := os.MkdirAll(dir, 0755); err != nil {
```

Comment:
I think permission should also be 0600

Reply:
0600 will only allow for Read and Writes for the owner of that directory, if we opened up another terminal and tried to get content out of that directory, we'd get "Permission denied" as 6 for the owner would mean they can't invoke execute which seems to be needed for MkdirAll, let's meet in the middle, how about `0700`? Still then I even wonder if other users in the same group will be able to use the file as they'll have `0` for their access